src/transformers/trainer.py

```diff
@@ -2515,7 +2516,7 @@ def hyperparameter_search(
         backend: Optional[Union["str", HPSearchBackend]] = None,
         hp_name: Optional[Callable[["optuna.Trial"], str]] = None,
         **kwargs,
-    ) -> BestRun:
+    ) -> Union[BestRun, List[BestRun]]:
         """
         Launch an hyperparameter search using `optuna` or `Ray Tune` or `SigOpt`. The optimized quantity is determined
         by `compute_objective`, which defaults to a function returning the evaluation loss when no metric is provided,
```

Comment:
documentation should be updated

Reply:
update the hpo_train.md

Reply:
What I mean here is that the `Return` type in the docstring as well as the `directions` argument should also be updated/added to mention this! 😉